hazelcast/src/main/java/com/hazelcast/internal/monitors/PerformanceMonitor.java

```diff
@@ -16,111 +16,135 @@
+    private final List<PerformanceMonitorPlugin> staticTasks = new CopyOnWriteArrayList<PerformanceMonitorPlugin>();
+    private final boolean singleLine;
+
+    public PerformanceMonitor(NodeEngineImpl nodeEngine) {
+        this.nodeEngine = nodeEngine;
+        this.logger = nodeEngine.getLogger(PerformanceMonitor.class);
+        GroupProperties props = nodeEngine.getGroupProperties();
+        this.enabled = props.getBoolean(PERFORMANCE_MONITOR_ENABLED);
+        this.singleLine = !props.getBoolean(PERFORMANCE_MONITOR_HUMAN_FRIENDLY_FORMAT);
+    }
 
-    public PerformanceMonitor(HazelcastInstanceImpl hazelcastInstance) {
-        this.hazelcastInstance = hazelcastInstance;
-        this.node = hazelcastInstance.node;
-        this.operationService = node.nodeEngine.getOperationService();
-        this.logger = node.getLogger(PerformanceMonitor.class);
-        this.metricRegistry = hazelcastInstance.node.nodeEngine.getMetricsRegistry();
-        this.enabled = node.getGroupProperties().getBoolean(GroupProperty.PERFORMANCE_MONITOR_ENABLED);
-        this.humanFriendlyFormat = node.getGroupProperties().getBoolean(GroupProperty.PERFORMANCE_MONITOR_HUMAN_FRIENDLY_FORMAT);
-        this.performanceLogFile = new PerformanceLogFile(this);
-        this.monitorThread = initMonitorThread();
+
+    List<PerformanceMonitorPlugin> getStaticTasks() {
+        return staticTasks;
     }
 
-    private MonitorThread initMonitorThread() {
+    /**
+     * Registers a MonitorTask to it will be scheduled.
+     *
+     * This method is threadsafe.
+     *
+     * There is no checking for duplicate registration.
+     *
+     * If the PerformanceMonitor is disabled, the call is ignored.
+     *
+     * @param performanceMonitorPlugin the monitorTask to register
+     * @throws NullPointerException if monitorTask is null.
+     */
+    public void register(PerformanceMonitorPlugin performanceMonitorPlugin) {
+        checkNotNull(performanceMonitorPlugin, "monitorTask can't be null");
+
         if (!enabled) {
-            return null;
+            return;
+        }
+
+        long periodMillis = performanceMonitorPlugin.getPeriodMillis();
+
+        logger.finest(performanceMonitorPlugin.getClass().toString() + " is " + (periodMillis == 0 ? "disabled" : "enabled"));
+
+        if (periodMillis == 0) {
+            // the task is disabled.
+            return;
         }
 
-        HazelcastThreadGroup threadGroup = node.getHazelcastThreadGroup();
-        int delaySeconds = node.getGroupProperties().getSeconds(GroupProperty.PERFORMANCE_MONITOR_DELAY_SECONDS);
+        performanceMonitorPlugin.setSingleLine(singleLine);
+        performanceMonitorPlugin.onStart();
 
-        return new MonitorThread(threadGroup, delaySeconds);
+        if (periodMillis > 0) {
+            // it is a periodic task
+            scheduler.scheduleAtFixedRate(new MonitorTaskRunnable(performanceMonitorPlugin), 0, periodMillis, MILLISECONDS);
+        } else {
```

Comment:
This looks nice and clean, but in fact every negative `periodMillis` will execute the plugin once, although it's documented to be `-1` (aka `RUN_ONCE`). Maybe we should check with `else if (periodMillis == RUN_ONCE)` here and throw an `IllegalArgumentException` for all other negative values? Don't know what our policy for wrong configurations is (if there is any).